### DISCUSSION TITLE
RateLimiter: needs to allow one invocation of method each 10 second or rare.

### DISCUSSION THREAD

**Discussion by YaraslauBarysenka** [resilience4j/resilience4j#2251] [Q&A]
Task: allow to send a feign client **getData** request each **10** second, otherwise feign.FeignException.TooManyRequests occurs.
Setup: Java 21, spring-boot-3.2, resilience4j-spring-boot3 2.1.0 version.

I use default resilience4j-spring-boot3 autoconfiguration.

application.yaml:
![image](https://github.com/user-attachments/assets/9c81d5ee-c7f5-4388-9ec0-127c2926d03c)

Feign client:
```
@FeignClient
public interface DataFeignClient {

    @RateLimiter(name = "getData")
    @GetMapping("/data")
    DataDto getData();
}
```
    
Finaly, we have a Job which use  DataFeignClient:
```
@Component
public class Job {
    private final DataFeignClient dataFeignClient;
    
    @Scheduled(initialDelayString = "", fixedDelayString = "")
    public void process() {
        for (int i = 0; i < 5; i++) {
            dataFeignClient.getData();
        }
    }
}
```

As result, we have just one thread, which perform ` dataFeignClient.getData()` function.
After that I run the application and face the issue.
Let's look at the time when ` dataFeignClient.getData()` function was performed:
```
2024-12-05T16:00:04.009 [scheduling-3] [DataFeignClient] DEBUG [DataFeignClient#getData] ---> GET ..
2024-12-05T16:00:06.955 [scheduling-3] [DataFeignClient] DEBUG [DataFeignClient#getData] ---> GET ..
2024-12-05T16:00:16.965 [scheduling-3] [DataFeignClient] DEBUG [DataFeignClient#getData] ---> GET ..
2024-12-05T16:00:26.962 [scheduling-3] [DataFeignClient] DEBUG [DataFeignClient#getData] ---> GET ..
2024-12-05T16:00:36.954 [scheduling-3] [DataFeignClient] DEBUG [DataFeignClient#getData] ---> GET ..
```

Time range is around **3** second between first and second execution (16:00:06.955 minus 16:00:04.009) , other Time ranges are **10** second.

Could you please help me how to fix it?

**Comment by tsrshk** [comment]
Thanks to your question! This question is important to my task as well

**Comment by RobWin** [comment]
You are not using https://github.com/resilience4j/resilience4j/tree/master/resilience4j-feign or?
you need to enhance feign with the decorators.
Annotations don't really work together with feign.
